uxarray/grid.py

```diff
@@ -158,35 +172,43 @@ def __from_vert__(self):
                 "face_node_connectivity": "Mesh2_face_nodes",
                 "face_dimension": "nMesh2_face"
             })
-        self.ds.Mesh2.attrs['topology_dimension'] = self.vertices[0].size
+        self.ds.Mesh2.attrs['topology_dimension'] = dataset.ndim
 
         # set default coordinate units to spherical coordinates
         # users can change to cartesian if using cartesian for initialization
         x_units = "degrees_east"
         y_units = "degrees_north"
-        if self.vertices[0].size > 2:
+        if dataset[0][0].size > 2:
             z_units = "elevation"
-
-        x_coord = self.vertices.transpose()[0]
-        y_coord = self.vertices.transpose()[1]
-        if self.vertices[0].size > 2:
-            z_coord = self.vertices.transpose()[2]
-
-        # single face with all nodes
-        num_nodes = x_coord.size
-        connectivity = [list(range(0, num_nodes))]
-
-        self.ds["Mesh2_node_x"] = xr.DataArray(data=xr.DataArray(x_coord),
+        x_coord = dataset[:, :, 0].flatten()
+        y_coord = dataset[:, :, 1].flatten()
+        if dataset[0][0].size > 2:
+            z_coord = dataset[:, :, 2].flatten()
+
+        # Identify unique vertices and their indices
+        unique_verts, indices = np.unique(dataset.reshape(
+            -1, dataset.shape[-1]),
+                                          axis=0,
+                                          return_inverse=True)
```

Comment:
Fix the formatting/indexing here, maybe consider the reshape right above the function call?